src/plugins/utils/tree-clusters.spec.ts

```diff
@@ -0,0 +1,797 @@
+                    name: 'ThHEri',
+                    type: 'wnTOxfON',
+                    children: [
+                        {
+                            start: 4.508507047607622,
+                            end: 4.5510423177725645,
+                            duration: 0.04253527016494285,
+                            name: 'vxNOUnENFDQE',
+                            type: 'wnTOxfON',
+                            children: [],
+                        },
+                        {
+                            start: 4.588760367894359,
+                            end: 9.848699863262404,
+                            duration: 5.2599394953680445,
+                            name: 'gGUATPcziMM',
+                            type: 'wnTOxfON',
+                            children: [],
+                        },
+                    ],
+                    level: 0,
+                    index: 2,
+                },
+                start: 4.508507047607622,
+                end: 4.5510423177725645,
+                duration: 0.04253527016494285,
+                name: 'vxNOUnENFDQE',
+                type: 'wnTOxfON',
+                children: [],
+                level: 1,
+                index: 3,
+            },
+        ],
+    },
+    {
+        color: void 0,
+        start: 4.588760367894359,
+        end: 9.848699863262404,
+        duration: 5.2599394953680445,
+        type: 'wnTOxfON',
+        level: 1,
+        nodes: [
+            {
+                parent: {
+                    parent: null,
+                    start: 4.508069066201776,
+                    end: 9.859908953776461,
+                    duration: 5.351839887574685,
+                    name: 'ThHEri',
+                    type: 'wnTOxfON',
+                    children: [
+                        {
+                            start: 4.508507047607622,
+                            end: 4.5510423177725645,
+                            duration: 0.04253527016494285,
+                            name: 'vxNOUnENFDQE',
+                            type: 'wnTOxfON',
+                            children: [],
+                        },
+                        {
+                            start: 4.588760367894359,
+                            end: 9.848699863262404,
+                            duration: 5.2599394953680445,
+                            name: 'gGUATPcziMM',
+                            type: 'wnTOxfON',
+                            children: [],
+                        },
+                    ],
+                    level: 0,
+                    index: 2,
+                },
+                start: 4.588760367894359,
+                end: 9.848699863262404,
+                duration: 5.2599394953680445,
+                name: 'gGUATPcziMM',
+                type: 'wnTOxfON',
+                children: [],
+                level: 1,
+                index: 4,
+            },
+        ],
+    },
+];
+
+describe('tree clusters', () => {
```

Comment:
Might be preferable to describe smaller data for tests? I think it's too hard to maintain. Also, it would be nice to be able to add tests for edge cases, and without a simple structure, this is very difficult to do. But for first time it is ok.)

Reply:
yeah, tests need to be more refined here
for now, they serve as some kind of protection against regressions, but not fully
I am back from vacation, so might work on that next these days to come up with something more maintainable with more coverage

Reply:
By the way, have you begun to clearly understand the whole project and clustering in particular? I think it will help you in future work with flame-chart.)

Reply:
Yes, you're right.) Good to hear. 

Cool, I think you will soon understand what you want. You can also write something about the plugins you want to implement in the roadmap.

Reply:
Currently at my work we have a sidebar with all the data as a list, and when we click on one of the entry it shows in the list only those with that name and highlights all of the nodes in the flamechart also
we don't have the outline, what we do we whiten out all of the nodes and preserve the color only on nodes that are selected
then when we select a specific one from that list, flamechart zooms in and whitens out everything except that one in the list that is clicked. Same is the other way around, if you click on the node in flamechart, it shows you in the list.
![tracer-all-selected](https://user-images.githubusercontent.com/31451544/162983381-95a9a25c-716b-4318-80cf-a0d2ee433c83.png)

Not sure if that is feasible currently as a  custom made plugin passed in from outside to a flamechart